app/models/grda_warehouse/import_threshold.rb

```diff
@@ -0,0 +1,189 @@
+    #   minimum count threshold and the percentage change threshold, otherwise `false`.
+    #
+    # The method performs the following checks:
+    # - Returns `false` if any of the required threshold values (`record_count_change_min_threshold`
+    #   or `record_count_change_percent_threshold`) or input parameters (`total` or `count`) are missing.
+    # - Returns `false` if `count` or `total` is zero (to avoid division errors and meaningless calculations).
+    # - Returns `false` if `count` is below the `record_count_change_min_threshold`.
+    # - Returns `false` if the percentage change is below the `record_count_change_percent_threshold`.
+    # - Returns `true` if all conditions are met.
+    #
+    def record_count_threshold_reached?(total, count)
+      return false unless record_count_change_min_threshold && record_count_change_percent_threshold && total && count
+      return false if count.zero? || total.zero?
+      return false if count < record_count_change_min_threshold
+
+      percent_change = (count / total.to_f) * 100
+      return false if percent_change < record_count_change_percent_threshold
+
+      true
+    end
+
+    ##
+    # Determines whether the count of errors during an import meets or exceeds
+    # the defined thresholds for triggering a notification.
+    #
+    # @param total [Integer] The total number of records in the import.
+    # @param count [Integer] The number of records containing an error.
+    # @return [Boolean] Returns `true` if the count meets or exceeds both the
+    #   minimum count threshold and the percentage change threshold, otherwise `false`.
+    #
+    # The method performs the following checks:
+    # - Returns `false` if any of the required threshold values (`error_count_min_threshold`
+    #   or `error_percent_threshold`) or input parameters (`total` or `count`) are missing.
+    # - Returns `false` if `count` or `total` is zero (to avoid division errors and meaningless calculations).
+    # - Returns `false` if `count` is below the `error_count_min_threshold`.
+    # - Returns `false` if the percentage change is below the `error_percent_threshold`.
+    # - Returns `true` if all conditions are met.
+    #
+    def error_count_threshold_reached?(total, count)
+      return false unless error_count_min_threshold && error_percent_threshold && total && count
+      return false if count.zero? || total.zero?
+      return false if count < error_count_min_threshold
+
+      percent_change = (count / total.to_f) * 100
+      return false if percent_change < error_percent_threshold
+
+      true
+    end
+
+    # Gather users from notification configurations into 3 categories
+    # 1. Those that have notifications for errors
+    # 2. Those that have notifications for count changes
+    # 3. Those that have both
+    def send_status_notifications(import_log_id, error_threshold_met, record_count_threshold_met, paused)
```

Comment:
I think keyword args are a better choice than positional once you get past one or two args. That makes it less likely that someone might call it incorrectly.
```suggestion
    def send_status_notifications(import_log_id:, error_threshold_met:, record_count_threshold_met:, paused:)
```

Also this method is kind of long